kedro/extras/extensions/ipython.py

```diff
@@ -83,12 +83,56 @@ def reload_kedro(
         logger.info("Registered line magic '%s'", line_magic.__name__)  # type: ignore
 
 
-def load_ipython_extension(ipython):
-    """Main entry point when %load_ext is executed"""
+def load_ipython_extension(ipython):  # pylint: disable=unused-argument
+    """
+    Main entry point when %load_ext is executed.
+    IPython will look for this function specificially.
+    See https://ipython.readthedocs.io/en/stable/config/extensions/index.html
 
-    global default_project_path
+    This function is called when users do `%load_ext kedro.extras.extensions.ipython`.
+    When user use `kedro jupyter notebook` or `jupyter ipython`, this extension is
+    loaded automatically.
+    """
+    import json
 
-    ipython.register_magic_function(reload_kedro, "line", "reload_kedro")
+    from IPython.core.magic import register_line_magic
+    from IPython.core.magic_arguments import argument, magic_arguments, parse_argstring
+
+    @magic_arguments()
+    @argument(
+        "path", help=("Path to the project root directory."), nargs="?", default=None
+    )
+    @argument(
+        "--env",
+        default=None,
+        help=("Environment for the Kedro project."),
+    )
+    @argument(
+        "--extra_params",
+        type=json.loads,
+        default=None,
+        help=(
+            "Optional dictionary containing extra project parameters for underlying "
+            "KedroContext. If specified, will update (and therefore take precedence "
+            "over) the parameters retrieved from the project configuration."
+        ),
+    )
+    @register_line_magic(
+        "reload_kedro"
+    )  # Register the line magic's name as `reload_kedro`
+    def reload_kedro_line_magic(line):
```

Comment:
```suggestion
    @register_line_magic
    def magic_reload_kedro(line):
```
Would that work? Following the example of `magic_cool` in https://ipython.readthedocs.io/en/stable/api/generated/IPython.core.magic_arguments.html. It would be tidier if that does work, but maybe it's just sloppy documentation 😅 Maybe you don't even need `@register_line_magic` at all? 🤔

Reply:
I'm just skimming through the IPython source code now and saw this:
```
def real_name(magic_func):
    """ Find the real name of the magic.
    """
    magic_name = magic_func.__name__
    if magic_name.startswith('magic_'):
        magic_name = magic_name[len('magic_'):]
    return getattr(magic_func, 'argcmd_name', magic_name)
```
so I think indeed calling it `magic_reload_kedro` will magically work 🎉

Reply:
I've got this figured out now - will explain!

Reply:
Long story short, we now have a nice help message:
```
In [23]: %reload_kedro?
Docstring:
::

  %reload_kedro [-e ENV] [--extra_params EXTRA_PARAMS] [path]

The `%reload_kedro` IPython line magic. See
 https://kedro.readthedocs.io/en/stable/tools_integration/ipython.html for more.

positional arguments:
  path                  Path to the project root directory. If not given, use the previously setproject root.

optional arguments:
  -e ENV, --env ENV     Kedro configuration environment name. Defaults to `local`.
  --extra_params EXTRA_PARAMS
                        Optional dictionary containing extra project parameters for underlying KedroContext. If specified, will update (and therefore take precedence over) the parameters retrieved
                        from the project configuration.
File:      ~/kedro_stuff/kedro/kedro/extras/extensions/ipython.py
```